File: lib/ClientUdp/command/exit_handler.dart
import 'dart:io';

import 'command_handler.dart';
import '../client_state.dart';
import '../constants.dart';
import '../handshake_manager.dart';

class ExitHandler implements CommandHandler {
  @override
  void handle(String args, ClientState clientState, HandshakeManager handshakeManager) {
    print("Exiting...");
    clientState.running = false;
  }

  @override
  String getDescription() {
    return Constants.CMD_EXIT_DESC;
  }
}

File: UdpClientFactor/src/main/java/UdpChatClient/command/ExitHandler.java
package UdpChatClient.command;

import UdpChatClient.ClientState;
import UdpChatClient.Constants;
import UdpChatClient.HandshakeManager;

public class ExitHandler implements CommandHandler {

    @Override
    public void handle(String args, ClientState clientState, HandshakeManager handshakeManager) {
        System.out.println("Exiting...");
        clientState.setRunning(false);
    }

    @Override
    public String getDescription() {
        return Constants.CMD_EXIT_DESC;
    }
}
